modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockWhileCaseInsensitiveUserFalseTestCase.java

```diff
@@ -0,0 +1,119 @@
+    private ServerConfigurationManager configurationManager;
+    private RemoteUserStoreManagerServiceClient usmServiceClient;
+    private AuthenticatorClient authClient;
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init();
+        String carbonHome = Utils.getResidentCarbonHome();
+        configureServerWithRestart(carbonHome);
+
+        super.init();
+
+        usmServiceClient = new RemoteUserStoreManagerServiceClient(backendURL, sessionCookie);
+        createLockedUser(TEST_USER_1, new String[]{ROLE_ADMIN}, TEST_USER_1_PASSWORD);
+        authClient = new AuthenticatorClient(backendURL);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void atEnd() throws Exception {
+
+        log.info("Deleting the user : " + TEST_USER_1 + ".");
+        usmServiceClient.deleteUser(TEST_USER_1);
+
+        log.info("Replacing the default configurations.");
+        configurationManager.restoreToLastConfiguration(false);
+    }
+
+    @SetEnvironment(executionEnvironments = {ExecutionEnvironment.ALL})
+    @Test(groups = "wso2.is", description = "Check whether the user is locked under CaseInsensitiveUsername property " +
+            "is false.", expectedExceptions = LoginAuthenticationExceptionException.class)
+    public void testCaseInsensitiveUsernameFalseUserLocking() throws Exception {
+
+        log.info("Login attempt to " + TEST_USER_1 + " user from " + TEST_USER_2 + " user.");
+        authClient.login(TEST_USER_2, TEST_USER_1_PASSWORD, isServer.getInstance().getHosts().get("default"));
+    }
+
+    private void configureServerWithRestart(String carbonHome)
+            throws AutomationUtilException, XPathExpressionException, IOException {
+
+        File defaultTomlFile = getDeploymentTomlFile(carbonHome);
+        File configuredTomlFile = new File(getISResourceLocation() + File.separator + "identityMgt"
+                + File.separator + "case_insensitive_user_false.toml");
+
+        log.info("Applying configured toml file.");
+        configurationManager = new ServerConfigurationManager(isServer);
+        configurationManager.applyConfigurationWithoutRestart(configuredTomlFile, defaultTomlFile, true);
+        configurationManager.restartGracefully();
+        log.info("Toml configurations applied.");
+    }
+
+    private void createLockedUser(String userName, String[] roles, String password) {
+
+        log.info("Creating a locked user account.");
+        ClaimValue[] claimValues = new ClaimValue[1];
+        ClaimValue lockedClaim = new ClaimValue();
+        lockedClaim.setClaimURI(ACCOUNT_LOCK_CLAIM_URI);
+        lockedClaim.setValue(Boolean.TRUE.toString());
+        claimValues[0] = lockedClaim;
+
+        try {
+            usmServiceClient.addUser(userName, password, roles, claimValues, null, false);
+            log.info("Locked user account created.");
+        } catch (Exception e) {
```

Comment:
Shall we catch only the specific exception types.

Reply:
Updated.